### ISSUE TITLE
Beta distribution strange behaviour

### ISSUE THREAD

**Issue by r3v1** [pytorch/pytorch#49894]
## 🐛 Bug

`torch.distributions.beta.Beta` gives no domain error when parameters `concentration1` and `concentration0` (a.k.a `alpha` and `beta` respectively).

From [Wikipedia](https://en.wikipedia.org/wiki/Beta_distribution), parameters of the distribution must be real positives. In fact, the [PyTorch documentation](https://pytorch.org/docs/master/distributions.html#torch.distributions.beta.Beta.arg_constraints) refers to those constraints as `arg_constraints = {'CONCENTRATION0': GREATERTHAN(LOWER_BOUND=0.0), 'CONCENTRATION1': GREATERTHAN(LOWER_BOUND=0.0)}` and that's fine, but executing with negative numbers gives no error.

Maybe I've forgot something, but it's strange.

## To Reproduce

```
>>> import torch   
>>> m = torch.distributions.Beta(concentration1=-4, concentration0=-3.54)
>>> m.sample((5,))
tensor([0.5000, 0.5000, 0.5000, 0.5000, 0.5000])  # <-- Is this possible?
```

## Expected behavior

The expected behaviour should bethe same as using other libraries, like [`scipy`'s `beta`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.beta.html#scipy.stats.beta):

```
>>> from scipy.stats.distributions import beta
>>> beta.rvs(-4, -3.54, size=5)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/david/.local/lib/python3.8/site-packages/scipy/stats/_distn_infrastructure.py", line 966, in rvs
    raise ValueError("Domain error in arguments.")
ValueError: Domain error in arguments.
````

## Environment

```
PyTorch version: 1.7.0
Is debug build: True
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Arch Linux (x86_64)
GCC version: (GCC) 10.2.0
Clang version: 11.0.0
CMake version: version 3.19.1

Python version: 3.8 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.4
[pip3] torch==1.7.0
[pip3] torchvision==0.6.1
[conda] Could not collect
```

cc @fritzo @neerajprad @alicanb @vishwakftw @nikitaved

**Comment by mruberry** [comment]
Thank you for reporting this issue, @r3v1. I think we would accept a PR fixing this.

**Comment by vishwakftw** [comment]
Hi @r3v1, by default we do not check arguments. But if you do pass `validate_args=True` like so `torch.distributions.Beta(concentration1=-4, concentration0=-3.54, validate_args=True)`, then you will see an error.

Optionally, you could set argument validation to run by default using `torch.distribution.Distribution.validate_args = True`, then you will not be required to pass `validate_args=True` every time.

cc: @fritzo maybe we should set `validate_args=True` by default (referring to our older discussion about this).

**Comment by r3v1** [comment]
I see @vishwakftw ,  thanks. Anyway, and despite my ignorance about the implementation and the purpose of the `validate_args` argument, it seems strange to me not to check by default, I suppose it will be a matter of the design of the implementation.

**Comment by mruberry** [comment]
Thanks @vishwakftw! Closing this issue as its currently by design.

**Comment by fritzo** [comment]
This should be fixed by #48743. I believe #48743 is currently held up by internal facebook errors, and @neerajprad is on it.
